src/components/Image.ts

```diff
@@ -145,21 +151,44 @@ export class Image extends Component<ImageProps, ImageChild> {
 	/**
 	 * Instantiate this component from the XML in an existing DOCX file.
 	 */
-	static fromNode(node: Node): Image {
-		return new Image(
-			evaluateXPathToMap<ImageProps>(
-				`
-					map {
-						"type": ./@${QNS.w}type/string(),
-						"clear": ./@${QNS.w}clear/string(),
-						"title": ./${QNS.wp}inline/${QNS.wp}docPr/@name/string(),
-						"width": docxml:length(./${QNS.wp}inline/${QNS.wp}extent/@cx, 'emu'),
-						"height": docxml:length(./${QNS.wp}inline/${QNS.wp}extent/@cy, 'emu')
-					}
-				`,
-				node,
-			),
+	static fromNode(node: Node, { archive, relationships }: ComponentContext): Image {
+		// Important nodes
+		const inlineNode = evaluateXPathToFirstNode(`./${QNS.wp}inline`, node);
+		const picNode = evaluateXPathToFirstNode(
+			`./${QNS.a}graphic/${QNS.a}graphicData/${QNS.pic}pic`,
+			inlineNode,
 		);
+
+		const title = evaluateXPathToString(`./${QNS.wp}docPr/@name/string()`, inlineNode);
+
+		const width = emu(evaluateXPathToNumber(`./${QNS.wp}extent/@cx/number()`, inlineNode));
+		const height = emu(evaluateXPathToNumber(`./${QNS.wp}extent/@cy/number()`, inlineNode));
+
+		if (relationships === null) {
+			// Our simplified images are always expected to reference a relationship ID
+			throw new Error(
+				'Failed to load image. The image is referencing a relationship ID but RelationhipsXml is null in the context.',
+			);
+		}
+
+		const blipEmbedRel = evaluateXPathToString(
+			`${QNS.pic}blipFill/${QNS.a}blip/@${QNS.r}embed/string()`,
+			picNode,
+		);
+		const location = relationships.getTarget(blipEmbedRel);
+		const data = archive.readBinary(location);
+
+		const type = new Promise<FileMime>((resolve) =>
```

Comment:
Did you mean `type = Promise.resolve(getMimeTypeForUint8Array)`?

Or, since `type: FileMime | Promise<FileMime>`,  just avoid the promise altogether…

Reply:
I indeed did mean that 😄

I cannot avoid the promise because `fromNode` is not async